runtime/runtime_controller.cc

```diff
@@ -115,11 +113,7 @@ std::unique_ptr<RuntimeController> RuntimeController::Clone() const {
 }
 
 bool RuntimeController::FlushRuntimeStateToIsolate() {
-  // TODO(dkwingsmt): Needs a view ID here (or platform_data should probably
-  // have multiple view metrics).
-  return SetViewportMetrics(kFlutterImplicitViewId,
-                            platform_data_.viewport_metrics) &&
-         SetLocales(platform_data_.locale_data) &&
+  return SetLocales(platform_data_.locale_data) &&
```

Comment:
It looks like this is only used by `RuntimeController::LaunchRootIsolate`. Should we send the implicit view's viewport metrics here to maintain backwards compatibility?

Reply:
`PlatformData` now contains a full map from view ID to view metrics, and this function flushes all of them to Dart by adding a view with this view metrics.